persistent/Database/Persist/Types/Base.hs

```diff
@@ -483,3 +488,36 @@ instance Exception OnlyUniqueException
 data PersistUpdate = Assign | Add | Subtract | Multiply | Divide
                    | BackendSpecificUpdate T.Text
     deriving (Read, Show)
+
+newtype EscapedLikeText = EscapedLikeText { runWildcardEscape :: Char -> T.Text }
+#if __GLASGOW_HASKELL__ >= 861
+  deriving (Semigroup, Monoid) via (Char -> T.Text)
+#endif
+
+#if __GLASGOW_HASKELL__ < 861
+instance Semigroup EscapedLikeText where
+  EscapedLikeText f <> EscapedLikeText g = EscapedLikeText (\c -> f c <> g c)
+
+instance Monoid EscapedLikeText where
+  mempty = EscapedLikeText (\_ -> T.pack "")
+  mappend = (<>)
+#endif
```

Comment:
As above - rather than have the `deriving via` instance *and* the manual handwritten instances, let's just use the hand written instances for now. That'll remove 8 lines of code.